config/config.go

```diff
@@ -983,9 +932,43 @@ func (p PackageConfig) Equals(other PackageConfig) bool {
 	return reflect.DeepEqual(p, other)
 }
 
-// PackageReference contains the deconstructed parts of a package reference in the config.
-// Eg: ${packages.some-package}/path/a/b/c -> {"some-package", "/path/a/b/c"}.
-type PackageReference struct {
-	Package       string
-	PathInPackage string
+// LocalDataParentDirectory returns the folder that will contain the all packages of this type.
+// Ex: /home/user/.viam/packages/ml_models
+func (p *PackageConfig) LocalDataParentDirectory() string {
+	return filepath.Join(viamDotDir, packagesDir, p.packageDirectoryFromType())
+}
+
+// LocalDataDirectory returns the folder where the package should be extracted.
+// Ex: /home/user/.viam/packages/ml_models/orgid_ballClassifier_0.1.2
+func (p *PackageConfig) LocalDataDirectory() string {
+	return filepath.Join(p.LocalDataDirectory(), p.sanitizedName())
+}
+
+// LocalDownloadPath returns the file where the archive should be downloaded before extraction
+func (p *PackageConfig) LocalDownloadPath() string {
+	return filepath.Join(p.LocalDataParentDirectory(), fmt.Sprintf("%s.download", p.sanitizedName()))
+}
+
+// sanitizedName returns the package name for the symlink/filepath of the package on the system.
+func (p *PackageConfig) sanitizedName() string {
+	return fmt.Sprintf("%s-%s", strings.ReplaceAll(p.Package, string(os.PathSeparator), "-"), p.sanitizedVersion())
+}
```

Comment:
Swapped function name to be a noun rather than a verb because before reading deeply, I assumed that `SanitizeName` would modify name in place